pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -251,379 +150,103 @@ def test_file_decriptors(self):
         `Pulp #4073 <https://pulp.plan.io/issues/4073>`_
 
         Do the following:
+
         1. Check if 'lsof' is installed. If it is not, skip this test.
         2. Create and sync a repo.
         3. Run the 'lsof' command to verify that files in the
            path ``/var/lib/pulp/`` are closed after the sync.
         4. Assert that issued command returns `0` opened files.
         """
-        cfg = config.get_config()
-        client = api.Client(cfg, api.json_handler)
-        cli_client = cli.Client(cfg, cli.echo_handler)
+        cli_client = cli.Client(self.cfg, cli.echo_handler)
 
         # check if 'lsof' is available
-        if cli_client.run(('which', 'lsof')).returncode != 0:
-            raise unittest.SkipTest('lsof package is not present')
+        if cli_client.run(("which", "lsof")).returncode != 0:
+            raise unittest.SkipTest("lsof package is not present")
 
-        repo = client.post(RPM_REPO_PATH, gen_repo())
-        self.addCleanup(client.delete, repo['pulp_href'])
+        repo_api = RepositoriesRpmApi(self.client)
+        repo = repo_api.create(gen_repo())
+        self.addCleanup(repo_api.delete, repo.pulp_href)
 
-        remote = client.post(RPM_REMOTE_PATH, gen_rpm_remote())
-        self.addCleanup(client.delete, remote['pulp_href'])
+        remote_api = RemotesRpmApi(self.client)
+        remote = remote_api.create(gen_rpm_remote())
+        self.addCleanup(remote_api.delete, remote.pulp_href)
 
-        sync(cfg, remote, repo)
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
 
-        cmd = 'lsof -t +D {}'.format(MEDIA_PATH).split()
+        cmd = "lsof -t +D {}".format(MEDIA_PATH).split()
         response = cli_client.run(cmd).stdout
         self.assertEqual(len(response), 0, response)
 
-
-class SyncMutatedPackagesTestCase(unittest.TestCase):
```

Comment:
we need to convert this test case also

Reply:
'test_mutated_packages' in basicsynctestcase class